### ISSUE TITLE
Magnet Tool sloppy deformation.

### ISSUE THREAD

**Issue by beeheemooth** [opentoonz/opentoonz#2849]
![ezgif-1-ef3dd42e2a12](https://user-images.githubusercontent.com/25724133/67215179-b17b6000-f429-11e9-9754-d860deddd636.gif)

Why does the Magnet Tool visually deform vectors correctly, but when the button is released, such a disaster happens? When using the Selection Tool, this does not happen.

### System Information
* **OpenToonz Version**: all
* **Operating System**: Windows 8.1

**Comment by RodneyBaker** [comment]
Confirmed, although...
This may related more to the shapes in question being circles with very few control points rather than the Magnet Tool itself.

The issue with circles and circular shapes likely needs to be addressed in a separate PR.
This PR also should be tested further after current refinements to influence of tool size are merged.

@beeheemooth Try rectangles instead of circles and note the results.

Prehaps this PR can be renamed and edited to focus on the issue with circularity or this one closed and a new PR opened?

**Comment by RodneyBaker** [comment]
As a test of the theory that there is a suboptimum calculation of the end points of a spline circle versus other shapes...

If we draw a rectangle with the Geo Tool and reshape it into a circle... copying and recreating the setup used in this issues demonstration we will note that the shapes maintain correction deformation.

As such I would look to the ends of the splines/curves to determine how best to factor in BOTH ends of the spline when in a circular shape.

This same issue with line endings can be seen in other places where we make connections using the Tape Tool as well as when drawing using Snap option.  The biases of both ends are apparently not taken into account thereby leading to inadvertent shape deformation.

**Comment by RodneyBaker** [comment]
I should have thought to test the Ellipse.
It appears the ellipse shape does not have the same issue as the circle.
Perhaps that is a good workaround to preventing the circle deformations.

So there is my best guess at an interim workaround:
**Draw Ellipse shape with Shift key held down to create a proper circle shape that resists being incorrectly deformed.**

**Comment by beeheemooth** [comment]
@RodneyBaker  It doesn't matter if you deform ellipses or circles - the result is equally terrible. Small details that have any curvature are especially distorted. Pay attention to the number of vertices after deformation.
![ezgif-1-1febd2a8aee4](https://user-images.githubusercontent.com/25724133/67600691-460ff600-f77b-11e9-91b1-9081a6ffaa93.gif)
Circle with a single point (!) It is even impossible to edit. 
![ezgif-1-72bcb24583e5](https://user-images.githubusercontent.com/25724133/67600798-85d6dd80-f77b-11e9-9e1b-a9496238efad.gif)
I think the problem is that after releasing the button the vectors are simplified, it is really necessary to maintain the smoothness of the lines, but this process is uncontrollable. The user does not have the opportunity to choose the degree of simplification of the vectors, or at least turn it off to get the desired deformation result.

**Comment by RodneyBaker** [comment]
I'm not getting the craziness with ellipses, only circles.
Not sure what we are doing differently.

**Comment by melieconiek** [comment]
Yeah. I get this a lot with gradual curves. Characters eyes will tend to get these huge bows in them after being affected by the magnet tool. I've seen this, although less frequently, with the tweak tool too. It can happen occasionally after just drawing a stroke sometimes even.

@beeheemooth I think you're right, this is happening when the stroke is being re-interpolated. Don't know how this is set up with the magnet tool--maybe it could have an accuracy slider like the brush tool.

**Comment by RodneyBaker** [comment]
A purer form of test for the underlying problem might be:

Create a vector level.
Turn Snap on.
Draw circles loosing trying to target ending at the start of the line being drawn.
Do this many times.
Note the results.

**Comment by RodneyBaker** [comment]
Closing as the result of similar testing is much improved although can still be improved more.
This level of precision will likely be resolved when vector graphics in general is improved.

Here's a video that displays what I'm currently seeing with similiar test as @beeheemooth showed.

https://github.com/user-attachments/assets/c8c91f66-1cf6-4241-b0ef-d7345f3b9b0d
